### ISSUE TITLE
Improve error typings

### ISSUE THREAD

**Issue by sneko** [sindresorhus/execa#909]
Hi,

Currently I have to do:
```ts
        try {
          const cmdResult = await $({
            timeout: 2000,
          })`sleep 10`;
        } catch (error) {
          if (error instanceof Error && !!(error as any).shortMessage) {
            console.log((error as any).shortMessage);
          } else {
            throw error;
          }
        }
```

I think it would be easier if people could do something like `error instanceof ExecaCommonError` so they can handle the case properly.

It would require some change so errors are instanciated but it should not break any linter after this switch.

In the meantime I saw currently it's a type (https://github.com/sindresorhus/execa/blob/3ecd447a8d93e3eae69836f960fe9ed7b0d16bb6/index.d.ts#L772) and I wanted to use it to cast the type of my error, but this one is not exported.

Thank you,

**Comment by ehmicky** [comment]
Hi @sneko,

That's an interesting request, thanks for posting it. 

This is actually a little harder than it seems. I'm going to answer this thoroughly, but first, would it be possible to create a separate issue for the part related to TypeScript types? This is actually a little of a different topic, which might deserve its own discussion. Thanks!

**Comment by tuler** [comment]
I also wanted that to properly check the `exitCode` which is present in case of "ExecaReturnBase"

**Comment by ehmicky** [comment]
@tuler Thanks for chiming in.
Could you please open a separate issue and include a code example? Thanks.

**Comment by ehmicky** [comment]
This should be fixed by #911.

There is now an `ExecaError` class, so you can do:

```js
import {execa, ExecaError} from 'execa';

try {
  await execa(...);
} catch (error) {
  if (error instanceof ExecaError) {
    ...
  }
}
```

This is typed and can work as a type guard in TypeScript.

When using `execaSync(...)`, the `ExecaSyncError` class must be used instead.

This will be available in the next release, which might take a few more weeks.

**Comment by tuler** [comment]
> @tuler Thanks for chiming in. Could you please open a separate issue and include a code example? Thanks.

My use case is already covered by the new feature. Thanks

**Comment by ehmicky** [comment]
This feature has been just released in Execa 9.0.0. Please see (and share!) the [release post](https://medium.com/@ehmicky/execa-9-release-d0d5daaa097f) and the [changelog](https://github.com/sindresorhus/execa/releases/tag/v9.0.0).
